src/ol/source/Raster.js

```diff
@@ -230,6 +230,20 @@ class RasterSource extends ImageSource {
       wantedTiles: {}
     };
 
+    this.setAttributions(function(frameState) {
+      const attributions = [];
+      for (let index = 0; index < options.sources.length; index++) {
```

Comment:
```suggestion
      for (let index = 0, iMax = options.sources.length; index < iMax; ++index) {
```

Reply:
Added the suggested change and force-pushed to keep the history clean.